src/form_routes.py

```diff
@@ -33,6 +35,7 @@ class UpdateForm(SanicForm):
 
 @form_blueprint.route('/create', methods=['GET'])
 @login_required
+@credential_whitelist_check()
 async def create_link_form(request, user):
     form = CreateForm(request)
     content = f"""
```

Comment:
Would it be possible to move these HTML snippets into the templates as well? I haven't tested it myself but it looks like this should work:
```jinja
{% extends 'base.html' %}

{% block content %}
  <div class="container">
  <form action="" method="POST">
    <h1 id="form-header">Create a new link</h1>
    {{ '<br>'.join(form.csrf_token.errors) }}
    {{ form.csrf_token }}
    ...
    <li>{{ form.endpoint(size=20, placeholder="Endpoint") }}</li>
    ...
  </form>
{% endblock %}
```
and this usage:
```python
template_loader(template_file, form=form)
```

Reply:
That's what I was trying originally, but it kept throwing an invalid form error. I may have made a mistake somewhere, though.

Reply:
This seems to work for me:
https://github.com/vojta-janousek/link-shortener/compare/master...radektheloner:test
So maybe try that again and send me the exact error? I will help you to resolve it.

Reply:
That works, thanks! Changed for both forms.